app/scripts/controllers/permissions/methodMiddleware.js

```diff
@@ -25,6 +32,20 @@ export default function createMethodMiddleware ({
 
       case 'eth_requestAccounts':
 
+        if (isProcessingRequestAccounts) {
+          res.error = ethErrors.rpc.resourceUnavailable(
+            'Already processing eth_requestAccounts. Please wait.'
+          )
+          return
+        }
+
+        try {
+          isProcessingRequestAccounts = true
+          await getUnlockPromise()
+        } catch (_) {}
```

Comment:
I'm not sure why this `catch` block is here - it should be fine to just `await` the unlock promise I would think

Reply:
Yeah, it's totally fine. Just me being paranoid. I'm addressing this by `This Promise will never reject.` to the `getUnlockPromise` docstring.